internal/db/bootstrap_logs.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2025 Canonical.
+
+package db
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/servermon"
+)
+
+func (d *Database) AddBootstrapLog(ctx context.Context, jobId uuid.UUID, logLine string) (err error) {
+	const op = errors.Op("db.AddBootstrapLog")
+
+	if err := d.ready(); err != nil {
+		return errors.E(op, err)
+	}
+
+	// Get the current line number for this bootstrap job.
+	var currentLineNumber int
+	err = d.DB.WithContext(ctx).
+		Model(&dbmodel.BootstrapLog{}).
+		Where("job_id = ?", jobId).
+		Select("COALESCE(MAX(line_number), 0)").
+		Scan(&currentLineNumber).Error
+	if err != nil {
+		return errors.E(op, "failed to get current line number", err)
+	}
+
+	nextLineNumber := currentLineNumber + 1
```

Comment:
Something to keep in mind here is what happens if you call this method at the same time for the same job.
Because the primary key is `(job_id, line_number)` it will fail which is correct.